### ISSUE TITLE
[Bug]: IOS WalletConnect, connection, approval, transaction data is not sent from metamask or is very unstable

### ISSUE THREAD

**Issue by makinghappen** [MetaMask/metamask-mobile#8396]
### Describe the bug

When using Metamask mobile on IOS with Safari/Chrome/Firefox browsers connection, approval, transaction data is not sent from Metamask to dAPP or when sent is very unstable.

It looks like that Metamask is not running in the background on IOS, causing that it is not fast enough to send data to WalletConnect servers before it redirects user back to the browser. 

### Expected behavior

Metamask IOS should work like it does on Android.

### Screenshots/Recordings

Please see the videos where we have recorded the process, with all the faults happening:

does not connect:
https://github.com/MetaMask/metamask-mobile/assets/100151762/171386e7-a014-46f1-af66-17f22eb593a0

connects, transaction request is not shown:
https://github.com/MetaMask/metamask-mobile/assets/100151762/56a99214-09da-4512-b4fa-e0c5ca457c6e

connects, but transaction gets rejected:
https://github.com/MetaMask/metamask-mobile/assets/100151762/bfebd79d-187c-440e-b58e-04ee8491b479

connects, transaction goes through(a big luck, this happened only twice during the testing), straight away a second transaction did not go through during this test
https://github.com/MetaMask/metamask-mobile/assets/100151762/253b105e-ae15-4914-9dcb-5b529b3ae6e7

### Steps to reproduce

The process can be described like this:
1. User opens dApp ( [dodao.dev](https://dodao.dev ) in IOS browser, and uses one of the links to connect(we have tested all possible: metamask://( metamask://wc?uri=$encodedUrl ), wc: ($encodedUrl), and universal link (https://metamask.app.link/wc?uri=$encodedUrl), where $encodedUrl is the url coming from WalletConnect SDK.
2. User is redirected to Metamask, and then sometimes connection request is being shown, sometimes not.
3. Then if connection request is shown, user is redirected back to the browser and then dApp is connected(with a certain random luck)
4. If the app is connected, user triggers a transaction, then, when going to Metamask user confirms the transaction, and is redirected back to the dApp, which will most often will not get transaction approval status from Metamask.

### Version

7.14.0

### Build type

None

### Device

iPhone 12 mini IOS 17.2.1, iPhone 14 IOS 16.2

### Operating system

iOS

### Additional context

we are developing [dodao.dev ](https://dodao.dev) to work on all platforms, actually there is not so many projects working on IOS safari with WalletConnect and Metamask, so we could not find a working example to compare with.

### Severity

Metamask on IOS in unusable with the WalletConnect browser connection

**Comment by mominnawaf** [comment]
This is used to send connection request

```ts
import { MetaMaskSDK } from '@metamask/sdk'
import { logger } from '../../utilities'

const MetaMaskComponent = (): React.JSX.Element => {
  const handleConnectMM = async (): Promise<void> => {
    logger('Initiating MetaMask connection request', debugMode)
    await window.ethereum
      ?.request({
        method: 'eth_requestAccounts',
        params: []
      })
      .then((accounts) => {
        console.log(accounts)
      })
      .catch((err: any) => {
        logger('Error connecting to MetaMask connection request', debugMode)
      })
  }

  const installMetaMask = async (): Promise<void> => {
    logger("MetaMask isn't installed prompt to download", debugMode)
    const MMSDK = new MetaMaskSDK()
    const ethereum = MMSDK.getProvider()
    void ethereum?.request({ method: 'eth_requestAccounts', params: [] })
  }

  const checkMM = async (): Promise<void> => {
    if ((Boolean(window.ethereum?.isMetaMask))) {
      void handleConnectMM()
    } else {
      void installMetaMask()
    }
  }

  return (
      <button
        onClick={() => { void checkMM() }}
      >
      Connect
      </button>
    
  )
}

export default MetaMaskComponent

```

This is used to sign transaction

```ts
   const provider = new ethers.providers.Web3Provider(window?.ethereum)
   const signer = provider.getSigner()
    if (selectedCurrency.is_native) {
       const transaction = {
            to: selectedCurrency?.receiver_address,
            value: utils.toWei(selectedCurrency.amount.raw_value_uint, 'wei')
         }
    const txResponse = await signer.sendTransaction(transaction)
    handlePendingPayment('crypto', txResponse.hash, wallet.accounts[0])
```

PS: The above code works perfectly in the case of desktop. On mobile its flaky and sometime takes longer to get a call back.

**Comment by mominnawaf** [comment]
Similar observation when used `@metamask/sdk-react`.

Followed: https://github.com/MetaMask/metamask-sdk/blob/main/packages%2Fexamples%2Fcreate-react-app%2Fsrc%2Findex.tsx

**Comment by christopherferreira9** [comment]
Hi @makinghappen ! We are currently updating the WalletConnect SDK and it should be released by the end of the month. We will update this issue when it gets released. 

@mominnawaf can you please open an issue on the [metamask-sdk](https://github.com/MetaMask/metamask-sdk) repository in order for us to debug it separately?

**Comment by christopherferreira9** [comment]
Hi @makinghappen ! We've merged the fixes for the daly in showing the request to the user once MetaMask opens. It'll be available in version `7.21.0`. I'm gonna close this issue for now but please feel free to open a new one if you see the same behaviour happening with this new version.
